juriscraper/opinions/united_states/state/ill.py

```diff
@@ -1,61 +1,80 @@
+   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+   2021-11-02: Status validation based on Amendment to Rule 23 rulings
+               https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
+"""
 
-from juriscraper.OpinionSite import OpinionSite
-from juriscraper.lib.string_utils import convert_date_string
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.html_utils import (
+    get_row_column_links,
+    get_row_column_text,
+)
+import datetime
+import re
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super(Site, self).__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.url = "http://www.illinoiscourts.gov/Opinions/recent_supreme.asp"
-        # Sometimes court forgets pdf link in 5th cell.
-        # So, only process rows with a link in 5th cell.
-        limiter = 'contains(./td[5]//a[1]/@href, "pdf")'
-        self.cell_path = '//table[@id="decisions"]//tr[' + limiter + "]/td[%d]"
+        self.url = (
+            "https://www.illinoiscourts.gov/top-level-opinions?type=supreme"
+        )
 
-    def _get_download_urls(self):
-        path = "%s//a[1]/@href" % (self.cell_path % 5)
-        return [href for href in self.html.xpath(path)]
+    def _process_html(self):
+        date_amend_rule_23 = datetime.date(2021, 1, 1)
+        for row in self.html.xpath("//table[@id='ctl04_gvDecisions']//tr"):
+            cells = row.xpath(".//td")
+            if len(cells) == 7:
+                try:
+                    name = get_row_column_text(row, 1)
+                    citation = get_row_column_text(row, 2)
+                    date = get_row_column_text(row, 3)
+                    url = get_row_column_links(row, 1)
+                    # After 2021-01-01, all documents are Precedential
+                    if (
+                        datetime.datetime.strptime(
+                            date.strip(), "%m/%d/%Y"
+                        ).date()
+                        >= date_amend_rule_23
+                    ):
+                        status = "Published"
+                    # Before 2021-01-01, Rule 23 rulings aren't Precedential
+                    else:
+                        decision_type = get_row_column_text(row, 5)
+                        if decision_type == "Rule 23":
+                            status = "Unpublished"
+                        else:
+                            status = "Published"
+                except IndexError:
+                    # If the opinion file's information is missing (as with
+                    # links to withdrawn opinions), skip record
+                    continue
+                docket = self.extract_docket(citation)
+                self.cases.append(
+                    {
+                        "date": date,
+                        "docket": docket,
```

Comment:
I'm surprised this is `docket` and not `docket_number`.

Reply:
Super class **OpinionSiteLinear** includes mehod `_get_docket_numbers` to return atribute `docket_number` from  `docket`:
https://github.com/freelawproject/juriscraper/blob/46d5fdb581c27512039bc23d10919418a2bab361/juriscraper/OpinionSiteLinear.py#L49-L52